### PR TITLE
[MNT] Proposed aeon deprecation policy

### PR DESCRIPTION
**Pull request by TonyBagnall**
further to the developers meeting on 5th September, those present agreed on trying to re-instate a deprecation policy starting from 0.6.0. This is a draft which I think reflects what we agreed, please feel free to adjust/add/remove.

### CODE REVIEW

**Comment by TonyBagnall** [comment]
I think we should have an exclusion for deprecation policy for any module deemed "experimental". Im thinking the new similarity search and condensing modules

**Review by baraline — APPROVED** [review]
The description of what should be deprecated and for how long is clear to me. The exclusion of experimental modules seems reasonable and is well argumented

**Review by dguijo — DISMISSED** [review]
lgtm!

**Review by hadifawaz1999 — APPROVED** [review]
LGTM !

**Review by MatthewMiddlehurst — CHANGES_REQUESTED** [review]
I will take a closer look tomorrow. The swap to .md seems to have had some casualties

https://aeon-toolkit--723.org.readthedocs.build/en/723/developer_guide/deprecation.html

I can go through and fix the formatting if you want

File: docs/developer_guide/deprecation.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+.. _developer_guide_deprecation:
+
+===========
+Deprecation
+===========
+
+Deprecation policy
+==================
```
[the remark this answers is not in the document]
Comment (TonyBagnall) [reply docs/developer_guide/deprecation.md]: nope, have removed one

**Review by MatthewMiddlehurst — DISMISSED** [review]
So, I think we should add a bit more to this (#840). However, now that the formatting issues are fixed by the look of things, I don't mind this going in first. Up to you.

**Review by MatthewMiddlehurst — APPROVED** [review]
Happy with the changes. 

@dguijo @baraline @hadifawaz1999 There are a few differences if you want to look again, but the times when you should deprecate and the experimental modules part have remained the same.

Main differences: 
- Can't deprecate from `0.6.1` to `0.7.0`, must do `0.6.1` to `0.8.0`
- Instructions for deprecating keyword arguments/parameters, only need to raise a warning when they are used rather than one for the whole class
- More examples

**Review by hadifawaz1999 — APPROVED** [review]
new changes LGTM !

**Pull request merged**